### PR TITLE
Check column count

### PR DESCRIPTION
**Pull request by ddesvillechabrol**

### CODE REVIEW

File: mergegi/mergegi.py [hunk 1]
```diff
@@ -19,8 +32,14 @@ def mergegi(barcodes_csv, raw_data_dir, merged_data_dir, paired=False, merge_lan
     file_to_merge = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
     with open(barcodes_csv) as csvfile:
         csv_reader = csv.reader(csvfile, skipinitialspace=True)
-        # Skip header
-        next(csv_reader)
+        try:
+            # skip header and check if column count is good
+            samplename, barcode, barcode2, project, lane = next(csv_reader)
+        except ValueError:
+            msg = ("Samplesheet is malformed. You must have samples names, barcode1, barcode2,"
```
Comment (khourhin) [inline mergegi/mergegi.py:39]: * You must have **sample** names ...

File: tests/resources/missing_column.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+samplename,barcode,project,lane
+C,42,tata,1
+B,42,toto,2
```
Comment (khourhin) [inline tests/resources/missing_column.csv:3]: No newline at the end of the file, not sure it is a problem or not...

**Review by khourhin — APPROVED** [review]

File: mergegi/mergegi.py (hunk 1 above)
Comment (ddesvillechabrol) [reply mergegi/mergegi.py:39]: Damned !

Comment (cokelaer) [inline mergegi/mergegi.py:39]: Your sample sheet seems to be ill-formed. You must have 5 columns corresponding to: ....lanes. They are expected to be found in this order.

**Review by cokelaer — APPROVED** [review]

**Pull request closed without merging**